src/utils/selectors.ts

```diff
@@ -210,6 +215,9 @@ export const maybeArraySelector = <T extends SparkleSelector<U[]>, U>(
   selector: SparkleSelector<U[]>
 ) => (ifTrue ? selector : emptyArraySelector);
 
+export const chatUIStateSelector: SparkleSelector<ChatState> = (state) =>
+  state.chat;
```

Comment:
Since everything here is a 'state selector' by definition, would something like this work better?

```suggestion
export const chatUISelector: SparkleSelector<ChatState> = (state) =>
  state.chat;
```